app/controllers/charges_controller.rb

```diff
@@ -9,25 +9,36 @@ def new
   end
 
   def create
+    # I know this controller action is fat. It will be refactored in a future PR
     order = current_user.orders.pending.last
     if order
       # Amount in cents
       amount = order.stripe_total
-      customer = Stripe::Customer.create(
-        email:  current_user.email,
-        source: params[:stripeToken]
-      )
+
+      if session[:stripeToken]
+        customer = Stripe::Customer.create(
+          email:  current_user.email,
+          source: session[:stripeToken]
+        )
+        session[:stripeToken] = nil
+        customer_id = customer.id
+      else
+        customer_id = current_user.stripe_token
+      end
+
+      current_user.save_stripe_token(customer_id)
 
       Stripe::Charge.create(
-        customer:     customer.id,
+        customer:     customer_id,
         amount:       amount,
         description:  'The Beautiful Rails Bookstore Purchase',
         currency:     'usd'
       )
 
-      order.order_status = OrderStatus.find_by(name: "Payment Complete") # Temporary solution
+      order.change_order_status_to("Payment Complete")
```

Comment:
Sounds good, it might clean up and make the code a bit more readable. Here is a gem that helps if you decide to go that route. https://github.com/aasm/aasm